bijection-util/src/main/scala/com/twitter/bijection/twitter_util/JavaFutureToTwitterFutureConverter.scala

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.bijection.twitter_util
+
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{ Future => JFuture }
+
+import com.twitter.util.{ Future, Promise, Try }
+
+import scala.annotation.tailrec
+
+/**
+ * Utility class for converting Java futures to Twitter's
+ *
+ * @param waitTimeMs Time spent sleeping by the thread converting java futures to
+ *                   twitter futures when there are no futures to convert in ms
+ */
+private[twitter_util] class JavaFutureToTwitterFutureConverter(waitTimeMs: Long = 1000L) {
```

Comment:
thread per future seems like a bad idea to me.

I'm not crazy about the polling all the time, but maybe the current approach is equivalent to the timer anyway (assuming a well implemented `Timer`).

We could do both approaches:
```scala
abstract class JavaFutureConverter {
  def apply[T](j: JFuture[T]): Future[T]
}

class FuturePoolJConverter(implicit fp: FuturePool) {
  ...
}

class TimerJConverter(t: Timer, d: ...) {

}
```

Then the `bijection` takes an implicit JavaFutureConverter which in some cases might be FuturePool based, in others timer/polling?